apps/beeai-server/src/beeai_server/domain/model.py

```diff
@@ -99,19 +102,26 @@ class NodeJsProvider(ManagedProvider):
     @field_validator("package", mode="after")
     @classmethod
     def _validate_package(cls, value: str, info: ValidationInfo) -> str:
-        base_file_path = info.data["base_file_path"]
+        base_file_path = info.data.get("base_file_path", None)
         if value.startswith((".", "/")) and base_file_path:
             package_path = Path(value)
             base_file_path = Path(base_file_path)
-            if base_file_path.name.endswith(".yaml"):
-                base_file_path = base_file_path.parent
             if not package_path.is_absolute():
                 return f"{base_file_path / package_path}"
         return value
 
     @asynccontextmanager
-    async def mcp_client(self) -> McpClient:  # noqa: F821
-        async with super()._get_mcp_client(command=["npx", "-y", self.package, *self.command]) as client:
+    @inject
+    async def mcp_client(self, configuration: Configuration) -> McpClient:  # noqa: F821
+        try:
+            github_url = GithubUrl.model_validate(self.package)
+            repo_path = await download_repo(configuration.cache_dir / "github_npm", github_url)
+            package_path = repo_path / (github_url.path or "")
+            await anyio.run_process(["npm", "install"], cwd=package_path)
```

Comment:
This is ugly, `npm/x` with tsx and filesystem/github dependencies drives me crazy. To download the repository and install dependencies locally is only way how I managed to make it work